### PR TITLE
Debugger fixes and improvements

### PR DESCRIPTION
**Pull request by joao-p-marques**
1. Map Python debugger to odoo/auto: All Odoo addons and core code are now mapped individually with the Python debugger.
2. Enable logs for test task when not in debug mode
3. Improve error messages

Error handling can still be improved, by trying to integrate better with VSCode tasks errors, but I had no success so far.

TT26728

### CODE REVIEW

**Review by Yajo — CHANGES_REQUESTED** [review]

**Comment by joao-p-marques** [comment]
Changes made @Yajo 

I had to test other addons to include other cases, but I tried to keep it simple and light, and test and install tasks are now tested together.

I also ended up ignoring the return code from the tests, so that our CI never fails because of some bad integration errors upstream, as has happened with https://github.com/Tecnativa/doodba-copier-template/runs/1389417193?check_suite_focus=true#step:14:988

File: tests/test_tasks_downstream.py [hunk 1]
```diff
@@ -177,75 +177,48 @@ def test_install(
...
+                assert "Executing odoo --stop-after-init --init utm" in stdout
+                assert _install_status("mail") == "installed"
+                assert _install_status("utm") == "installed"
+                assert _install_status("note") == "uninstalled"
+            # Test "note" simple call in init mode (default)
+            stdout = invoke("test", "-m", "note", "--mode", "init", retcode=None)
+            assert (
+                "Executing odoo --test-enable --stop-after-init --workers=0 -i note"
+                in stdout
             )
```
Comment (Yajo) [inline tests/test_tasks_downstream.py:201]: I think this'd be important:

```suggestion
            )
            assert "devel odoo.modules.loading: All post-tested" in stdout
```

File: tests/test_tasks_downstream.py [hunk 2]
```diff
@@ -177,75 +177,48 @@ def test_install(
...
-            invoke("test", "-m", "purchase")
-            stdout = _wait_for_test_to_start()
+            assert _install_status("note") == "installed"
+            # Test "note" simple call in update mode
+            stdout = invoke("test", "-m", "note", "--mode", "update", retcode=None)
             assert (
-                "Executing odoo --test-enable --stop-after-init --workers=0 -i purchase"
+                "Executing odoo --test-enable --stop-after-init --workers=0 -u note"
                 in stdout
             )
```
Comment (Yajo) [inline tests/test_tasks_downstream.py:208]: Here too:

```suggestion
            )
            assert "devel odoo.modules.loading: All post-tested" in stdout
```

**Review by Yajo — CHANGES_REQUESTED** [review]
Almost there

**Comment by Yajo** [comment]
Review the failure. Maybe the test is wrong or maybe odoo is really not testing the addons.

**Comment by joao-p-marques** [comment]
Yes, I am already on it. Seems like the correct log is
```
devel odoo.service.server: All post-tested
```
OTOH, I am also getting different logs in v14. I am investigating if this logs change between versions...

**Comment by Yajo** [comment]
> I am investigating if this logs change between versions...

Most probably yes, because I copied https://github.com/Tecnativa/doodba-copier-template/pull/176#discussion_r522024064 from v12.

**Comment by joao-p-marques** [comment]
Yes, I ended up creating the `_tests_ran` function to check specific logs base on the current version. Let's see if it works for all the versions.

**Review by Yajo — APPROVED** [review]

File: tests/test_tasks_downstream.py [hunk 3]
```diff
@@ -201,31 +195,28 @@ def test_install_test(
...
             # Test "note" simple call in init mode (default)
+            assert _install_status("note") == "uninstalled"
             stdout = invoke("test", "-m", "note", "--mode", "init", retcode=None)
             # Ensure "note" was installed and tests ran
             assert _install_status("note") == "installed"
             assert _tests_ran(stdout, supported_odoo_version, "note")
             # Test "note" simple call in update mode
             stdout = invoke("test", "-m", "note", "--mode", "update", retcode=None)
-            assert _tests_ran(stdout, supported_odoo_version, "note", mode="-u")
+            assert _tests_ran(stdout, supported_odoo_version, "note")
```
Comment (joao-p-marques) [inline tests/test_tasks_downstream.py:219]: If you are doing `assert`s in the _tests_ran function, this needs to be adapted. Either return true or false in the function, or just do a normal call here.

**Comment by joao-p-marques** [comment]
Just a small doubt... Otherwise tests look much better now, thank you!

**Pull request merged**